python/ray/ml/train/gbdt_trainer.py

```diff
@@ -0,0 +1,189 @@
+    _tune_callback_cls: type
+    _default_ray_params: Dict[str, Any] = {"checkpoint_frequency": 1}
+    _init_model_arg_name: str
+
+    def __init__(
+        self,
+        label_column: str,
+        params: Dict[str, Any],
+        datasets: Dict[str, GenDataset],
+        dmatrix_params: Optional[Dict[str, Dict[str, Any]]] = None,
+        scaling_config: Optional[ScalingConfig] = None,
+        run_config: Optional[RunConfig] = None,
+        preprocessor: Optional[Preprocessor] = None,
+        resume_from_checkpoint: Optional[Checkpoint] = None,
+        **train_kwargs,
+    ):
+        self.label_column = label_column
+        self.params = params
+        self.dmatrix_params = dmatrix_params or {}
+        self.train_kwargs = train_kwargs
+        super().__init__(
+            scaling_config, run_config, datasets, preprocessor, resume_from_checkpoint
+        )
+        self._validate_config_and_datasets()
+
+    def _validate_config_and_datasets(self) -> None:
+        if TRAIN_DATASET_KEY not in self.datasets:
+            raise KeyError(
+                f"'{TRAIN_DATASET_KEY}' key must be preset in `datasets`. "
+                f"Got {list(self.datasets.keys())}"
+            )
+        if self.dmatrix_params:
+            for key in self.dmatrix_params:
+                if key not in self.datasets:
+                    raise KeyError(
+                        f"`dmatrix_params` dict contains key '{key}' "
+                        f"which is not present in `datasets`."
+                    )
+
+    def _get_dmatrices(
+        self, dmatrix_params: Dict[str, Any]
+    ) -> Dict[str, xgboost_ray.RayDMatrix]:
+        return {
+            k: self._dmatrix_cls(
+                v, label=self.label_column, **dmatrix_params.get(k, {})
+            )
+            for k, v in self.datasets.items()
+        }
+
+    def _convert_scaling_config_to_ray_params(
+        self, scaling_config: ScalingConfig
+    ) -> xgboost_ray.RayParams:
+        scaling_config_dataclass = ScalingConfigDataClass(**scaling_config)
+        resources_per_worker = scaling_config_dataclass.additional_resources_per_worker
+        num_workers = scaling_config_dataclass.num_workers
+        cpus_per_worker = scaling_config_dataclass.num_cpus_per_worker
+        gpus_per_worker = scaling_config_dataclass.num_gpus_per_worker
+
+        ray_params = self._ray_params_cls(
+            num_actors=int(num_workers),
+            cpus_per_actor=int(cpus_per_worker),
+            gpus_per_actor=int(gpus_per_worker),
+            resources_per_actor=resources_per_worker,
+            **self._default_ray_params,
+        )
+
+        return ray_params
+
+    def _load_model_from_checkpoint(self):
+        raise NotImplementedError
+
+    def _train(self, **kwargs):
+        raise NotImplementedError
+
+    @property
+    def _ray_params(self) -> xgboost_ray.RayParams:
+        return self._convert_scaling_config_to_ray_params(self.scaling_config)
```

Comment:
Just a nit: but can we just store this as an attribute? do we need to do this conversion everytime we access self._ray_params?

Reply:
It's very cheap and guarantees consistency should scaling config be modified somewhere along the way